### PR TITLE
JIT: Add support for `SwiftIndirectResult` in Swift calling convention

### PR DESCRIPTION
**Pull request by jakobbotsch**
Add a test for pinvoke/reverse pinvokes involving `SwiftIndirectResult`, and add the necessary support to make the tests pass in RyuJIT.

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @amanasifkhalid 

FYI @jkurdek @kotlarmilos @matouskozak

File: src/coreclr/jit/importercalls.cpp [hunk 1]
```diff
@@ -2118,7 +2137,7 @@ void Compiler::impPopArgsForSwiftCall(GenTreeCall* call, CORINFO_SIG_INFO* sig,
 
     // If using SwiftError*, spill entire stack as we will need to reuse the
     // error argument after the call.
-    if (checkEntireStack)
+    if (spillStack)
```
Comment (amanasifkhalid) [inline src/coreclr/jit/importercalls.cpp:2140]: Could you update this comment to mention the SwiftIndirectResult case, please?

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

File: src/coreclr/jit/importercalls.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/importercalls.cpp:2140]: I moved the existing comment up to where we assign `spillStack` in the `SwiftError*` case instead (there is already a comment on the new `spillStack = true` I added)

**Pull request merged**